pyspectral/tests/test_rayleigh.py

```diff
@@ -280,6 +280,17 @@ def test_rayleigh_reduction(self, fake_lut_hdf5):
         retv = rayleigh.Rayleigh.reduce_rayleigh_highzenith(sun_zenith, in_rayleigh, 30., 90., 1.5)
         np.testing.assert_allclose(retv, TEST_RAYLEIGH_RESULT_R2)
 
+    def test_rayleigh_getname(self):
+        """Test logic for Rayleigh instrument selection."""
+        rayl = _create_rayleigh(platform='FY-4A')
+        assert rayl.sensor == 'agri'
+        rayl = _create_rayleigh(platform='FY-4B')
+        assert rayl.sensor == 'agri'
+        rayl = _create_rayleigh(platform='FY-4B', sensor='ghi')
+        assert rayl.sensor == 'ghi'
+        rayl = _create_rayleigh(platform='FY-4B', sensor='nosensor')
+        assert rayl.sensor == 'agri'
```

Comment:
Why does a non-existent sensor return `agri`?

Reply:
I've updated this now, so if the available sensors is a `list` but the user does not explicitly state which sensor they want then a `ValueError` will be raised.